### ISSUE TITLE
Thoughts about the current way of styling stuff in the boilerplate

### ISSUE THREAD

**Issue by mconiglio** [loopstudio/react-app-boilerplate#162]
We have discussed this with some members of the team, and came to the conclusion that not everyone is satisfied with the current approach of styling things.

**Keep in mind that this issue is not necessarily focused on finding a new solution for this, but on discussing whether we want to continue using it or not.**

**If we want to continue with the current approach, then great! (some people really like it actually)
If not, well... yes, let's discuss possible new solutions**

The current tools we are using are:
- Tailwind CSS
- CSS modules
- SASS

I think @mollerjorge mentioned that it's not a good thing to have to use classes this way `md\:py-2,` just because the `:` have to be escaped. There is a solution for that, that can be found [here](https://tailwindcss.com/docs/configuration/#separator). So we could change `:` to be something like `-` or some other character.

Despite that, my personal opinion is that css modules and utility first css libraries don't work great together. We are using the `@extend` keyword to compose utility classes, and a lot of those classes have only one property inside them (like `bg-white` -> `background-color: white`). So we are not taking most of the advantages that libraries like tailwind have to offer.

Personally I really like the utility first approach, and tailwind seems to be one of the best options for that. And when I saw his creator talk about the `apply`/`extend` usage, his recommendation for it was related to the times when you are not using a component focused library, to avoid repetition and increase reusability. But, when using something like react, the reusability should be focused on the components, not on css classes (which we aren't doing either because we have separate css files for each component).

So, to sum up, I think we should go with one or the two:

First option:
- Tailwind CSS
- Files with Custom utility classes following the tailwind naming conventions [example here](https://tailwindcss.com/docs/adding-new-utilities/#using-css)
- Some library like [classnames](https://github.com/JedWatson/classnames#readme) or [clsx](https://github.com/lukeed/clsx)

Second option:
- CSS modules
- (Maybe) SASS
- (Maybe) some ui library
